crates/derive/src/stages/batch_queue.rs

```diff
@@ -146,6 +150,9 @@ where
                     remaining.push(batch.clone());
                 }
                 BatchValidity::Drop => {
+                    // If we drop a batch, flush previous batches buffered in the BatchStream
+                    // stage.
+                    self.prev.flush();
```

Comment:
I _think_ this is right, but would be good to triple check. The stage should only be flushed on partial span batch invalidity, I wonder if there are other batch drop cases where this shouldn't be hit?

Reply:
My thinking:
- Pre-holocene this flush will be a no-op.
- Post-holocene, single batches are either actually single, or derived from a span batch.
  - If they are single, there shouldn't be a span batch in the batch stream due to strict ordering.
  - If they are derived from a span batch the whole span batch and it's channel need to be flushed.